libtcmu_config.c

```diff
@@ -130,25 +130,25 @@ do { \
 	free(cfg->key); \
 } while (0);
 
-#define TCMU_CONF_CHECK_LOG_LEVEL(key) \
-do { \
-	struct tcmu_conf_option *option; \
-	option = tcmu_get_option(#key); \
-	if (!option) \
-		return; \
-	if (option->opt_int > TCMU_CONF_LOG_LEVEL_MAX) { \
-		option->opt_int = TCMU_CONF_LOG_LEVEL_MAX; \
-	} else if (option->opt_int < TCMU_CONF_LOG_LEVEL_MIN) { \
-		option->opt_int = TCMU_CONF_LOG_LEVEL_MIN; \
-	} \
-} while (0);
-
 static void tcmu_conf_set_options(struct tcmu_config *cfg)
 {
 	/* set log_level option */
 	TCMU_PARSE_CFG_INT(cfg, log_level);
-	TCMU_CONF_CHECK_LOG_LEVEL(log_level);
-	tcmu_set_log_level(cfg->log_level);
+	if (cfg->log_level) {
+		tcmu_set_log_level(cfg->log_level);
+	}
+
+	/* set log_dir path option */
+	TCMU_PARSE_CFG_STR(cfg, log_dir_path);
+	/*
+	 * The priority of the logdir setting is:
+	 * 1, --tcmu_log_dir/-l LOG_DIR_PATH
+	 * 2, export TCMU_LOGDIR="/var/log/mychoice/"
+	 * 3, tcmu.conf
+	 * 4, default /var/log/
+	 */
+	if (!tcmu_get_log_dir())
```

Comment:
If someone has set tcmu_log_dir, then tries to reset it via the tcmu.conf reconfig method then this would silently ignore the request right? Maybe just add a warning about the operation failing due to it not being supported.

Reply:
Yes, then the tcmu.conf's will be ignored.

I'll add the warning messages.